lib/teiserver_web/views/api/o_auth/code_view.ex

```diff
@@ -0,0 +1,35 @@
+defmodule TeiserverWeb.OAuth.CodeView do
+  use TeiserverWeb, :view
+
+  def token(%{token: token}) do
+    expires_in = DateTime.diff(token.expires_at, DateTime.utc_now(), :second)
+
+    %{
+      access_token: token.value,
+      expires_in: expires_in,
+      refresh_token: token.refresh_token.value,
+      token_type: "Bearer"
+    }
+  end
+
+  def error(conn) do
+    Map.take(conn, [:error_description]) |> Map.put("error", "invalid_request")
+  end
+
+  def metadata(_) do
+    base = Application.fetch_env!(:teiserver, Teiserver.OAuth)[:issuer]
+
+    %{
+      issuer: base,
+      authorization_endpoint: base <> ~p"/oauth/authorize",
+      token_endpoint: base <> ~p"/oauth/token",
+      token_endpoint_auth_methods_supported: ["none", "client_secret_post"],
```

Comment:
https://www.rfc-editor.org/rfc/rfc6749.html#section-2.3.1

> The authorization server MUST support the HTTP Basic authentication scheme for authenticating clients that were issued a client password.

I interpret that as `client_secret_basic` effectively being required on this list.

Reply:
I've added support for basic auth in 19bc2c23d955ced5a6fab5f71d7fc0bc05e77bcd

Reply:
This seems to not be resolved, I've tested with curl and I get `missing client_id` when passing client_id via Basic auth:

```
curl --user generic_lobby: -d grant_type=authorization_code -s -X POST http://localhost:4000/oauth/token
```